### DISCUSSION TITLE
Anyway to change frame resolution or rotate captured screen?

### DISCUSSION THREAD

**Discussion by LichLord91** [LizardByte/community#81] [Sunshine]
I was able to successfully use cmake on my Steam Deck to create my own  package for sunshine so that I could use the wayland/kms session. However using kms/wayland the screen is captured vertically. The steam deck screen is naturally 800:1280  5:8 aspect ratio which valve rotates by 90 degrees to show a horizontal view to make it look 1280x800.  I was curious if there's a way to rotate/set the custom frame capture resolution through sunshine. 

game mode capture

![game_mode](https://user-images.githubusercontent.com/8435580/190379163-c71a6261-ff72-44f1-a8d9-34187375aa5e.png)

desktop mode capture 

![desktop_mode](https://user-images.githubusercontent.com/8435580/190379202-9f681a7b-48fb-4f35-b29a-d1c7c2de5660.PNG)

**Comment by LichLord91** [comment]
P.S happy to build a custom branch/modification of code to test this if there's a special version we can make for steam deck

**Comment by LichLord91** [comment]
Someone posted this last night funny enough, going to give it a try

https://clbin.com/2PSOM

```
diff --git a/src/platform/linux/kmsgrab.cpp b/src/platform/linux/kmsgrab.cpp
index 3b9257b..214e262 100644
--- a/src/platform/linux/kmsgrab.cpp
+++ b/src/platform/linux/kmsgrab.cpp
@@ -523,8 +523,8 @@ public:
         if(monitor != std::end(pos->crtc_to_monitor)) {
           auto &viewport = monitor->second.viewport;
 
-          width    = viewport.width;
-          height   = viewport.height;
+          width    = viewport.height; // owokitty mod: hardcoded swap height and width for Steam Deck
+          height   = viewport.width;  // owokitty mod: hardcoded swap height and width for Steam Deck
           offset_x = viewport.offset_x;
           offset_y = viewport.offset_y;
         }
```

https://discord.com/channels/352065098472488960/465369864534491146/1019952614353403914

![image](https://user-images.githubusercontent.com/8435580/190496576-a9b45a29-57fd-4811-8159-b0fe72438da6.png)

**Reply by lyndonguitar** [reply]
> Someone posted this last night funny enough, going to give it a try
> 
> https://clbin.com/2PSOM
> 
> ```
> diff --git a/src/platform/linux/kmsgrab.cpp b/src/platform/linux/kmsgrab.cpp
> index 3b9257b..214e262 100644
> --- a/src/platform/linux/kmsgrab.cpp
> +++ b/src/platform/linux/kmsgrab.cpp
> @@ -523,8 +523,8 @@ public:
>          if(monitor != std::end(pos->crtc_to_monitor)) {
>            auto &viewport = monitor->second.viewport;
>  
> -          width    = viewport.width;
> -          height   = viewport.height;
> +          width    = viewport.height; // owokitty mod: hardcoded swap height and width for Steam Deck
> +          height   = viewport.width;  // owokitty mod: hardcoded swap height and width for Steam Deck
>            offset_x = viewport.offset_x;
>            offset_y = viewport.offset_y;
>          }
> ```
> 
> https://discord.com/channels/352065098472488960/465369864534491146/1019952614353403914
> 
> ![image](https://user-images.githubusercontent.com/8435580/190496576-a9b45a29-57fd-4811-8159-b0fe72438da6.png)

have you ever solved it?

**Reply by LichLord91** [reply]
Yes 

https://gist.github.com/robertkirkman/753922262259486ec417e5ff8b5b924b

**Comment by ReenigneArcher** [comment]
Probably should move this discussion here: https://discord.com/channels/804382334370578482/1019471681913372672

Since that thread was started by the creator of the patch you are referring to.

https://app.lizardbyte.dev/discord

**Reply by backcountrymountains** [reply]
Any idea what I modify in the windows files to achieve this result?
Thanks.
